includes/embeds/class-amp-gallery-embed.php

```diff
@@ -129,25 +129,14 @@ public function shortcode( $attr ) {
 				'width'  => $width,
 				'height' => $height,
 				'alt'    => trim( wp_strip_all_tags( get_post_meta( $attachment_id, '_wp_attachment_image_alt', true ) ) ), // Logic from wp_get_attachment_image().
+				'id'     => $attachment_id,
 			];
 		}
 
 		$args = [
-			'images' => $urls,
+			'images'   => $urls,
+			'lightbox' => ! empty( $atts['lightbox'] ),
 		];
-		if ( ! empty( $atts['lightbox'] ) ) {
-			$args['lightbox'] = true;
-			$lightbox_tag     = AMP_HTML_Utils::build_tag(
-				'amp-image-lightbox',
-				[
-					'id'                           => AMP_Base_Sanitizer::AMP_IMAGE_LIGHTBOX_ID,
-					'layout'                       => 'nodisplay',
-					'data-close-button-aria-label' => __( 'Close', 'amp' ),
-				]
-			);
-			/* We need to add lightbox tag, too. @todo Could there be a better alternative for this? */
-			return $this->render( $args ) . $lightbox_tag;
-		}
```

Comment:
Now that this is using [amp-lightbox-gallery](https://amp.dev/documentation/components/amp-lightbox-gallery/) as [recommended by amphtml](https://github.com/ampproject/amphtml/issues/25021#issuecomment-541846264), all that's needed is a `lightbox` attribute on the `<amp-img>` or `<amp-carousel>`